### ISSUE TITLE
Let MajdataEdit to be compatible with Wine environment

### ISSUE THREAD

**Issue by aon3ko** [LingFeng-bbben/MajdataView#146]
## Where?

MajdataEdit

## New Feature Request

Sometimes we have to use old machine to daily-drive, but the Windows nowadays work poorly on old machines. Therefore, for me, there's a need to use Linux. Unfortunately, this software still not compatible with Wine, I hope that maybe we can solve the compatible issue together.

**Comment by aon3ko** [comment]
For more details, this file is the Terminal output when you execute MajdataEdit with a clean .wine environment.
Arch Linux w/ Linux 6.6.23, Wine 9.5
1st command execution, the initialization of Wine environment.
2nd command execution, the executable requested the installation of .NET 6.0 runtime.
3rd command execution, the installation of .NET 6.0.28 has being performed.
Finally, the output of `wine ./MajdataEdit.exe`

[MajdataEdit-4.3.1_Wine-9.5.txt](https://github.com/LingFeng-bbben/MajdataView/files/14856718/MajdataEdit-4.3.1_Wine-9.5.txt)

**Comment by LingFeng-bbben** [comment]
`//install wine and winetricks 
//install .net6.0
WINEPREFIX=~/.maj_test wine windowsdesktop-runtime-6.0.27-win-x64.exe 
//install grcko
wget https://dl.winehq.org/wine/wine-gecko/2.47.4/wine-gecko-2.47.4-x86_64.msi
//some fonts
WINEPREFIX=~/.maj_test winetricks corefonts
WINEPREFIX=~/.maj_test winetricks cjkfonts
WINEPREFIX=~/.maj_test winetricks richtx32
WINEPREFIX=~/.maj_test winetricks riched30
WINEPREFIX=~/.maj_test winetricks riched20
//dx to vulkan
WINEPREFIX=~/.maj_test winetricks dxvk
//remember to have software rendering
WINEPREFIX=~/.maj_test wine Majdata/MajdataEdit.exe `

**Comment by aon3ko** [comment]
> a working one, tested on debian 12 you may find the editor freeze when start playing I got no time to find this out so hopefully you can get some progress.
> 
> ```
> //install .net6.0
> WINEPREFIX=~/.maj_test wine windowsdesktop-runtime-6.0.27-win-x64.exe 
> //install grcko
> wget https://dl.winehq.org/wine/wine-gecko/2.47.4/wine-gecko-2.47.4-x86_64.msi
> //some fonts
> WINEPREFIX=~/.maj_test winetricks corefonts
> WINEPREFIX=~/.maj_test winetricks cjkfonts
> WINEPREFIX=~/.maj_test winetricks richtx32
> WINEPREFIX=~/.maj_test winetricks riched30
> WINEPREFIX=~/.maj_test winetricks riched20
> //dx to vulkan
> WINEPREFIX=~/.maj_test winetricks dxvk
> //remember to have software rendering
> WINEPREFIX=~/.maj_test wine Majdata/MajdataEdit.exe
> ```

Thank you for your reply. I found that the ```corefonts``` is the basic package to make this program operable. I'll try it with more time.

**Comment by XajilX** [comment]
Although I didn't know the reason, it seems that Edit works perfectly in Wine by disabling drawing FFT curve. 
I just simply commit out MainWindowCore.cs#L587 and compiled another one.

**Comment by XajilX** [comment]
Got a better workaround: just set `visualEffectRefreshTimer` to a larger number (eg. 16.66 for 60Hz)
Maybe it's just because the original timer setting is too fast for drawing UI in Wine.

**Comment by DavidScann** [comment]
So does the workaround require me to compile the program from source? That sounds like a bit of a pain; albeit a compromise that I may have to take.

**Comment by XajilX** [comment]
> So does the workaround require me to compile the program from source? That sounds like a bit of a pain; albeit a compromise that I may have to take.

Yes you need to compile from the source code, but I think it should be an easy task. I hope this could be an options in settings in the future. 

Additionally, I found that it's much easier to get crash in wine like #172 . So maybe you should use old version for recompile.

**Comment by LeZi9916** [comment]
> > So does the workaround require me to compile the program from source? That sounds like a bit of a pain; albeit a compromise that I may have to take.
> 
> Yes you need to compile from the source code, but I think it should be an easy task. I hope this could be an options in settings in the future. 
> 
> Additionally, I found that it's much easier to get crash in wine like #172 . So maybe you should use old version for recompile. 

We will restart the maintenance work on Edit/View in the near future, including cleaning up bad code, which may fix crashes like #172
